spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/handler/support/RoutePredicateFactoryUtils.java

```diff
@@ -34,4 +45,82 @@ public static void traceMatch(String prefix, Object desired, Object actual, bool
 			logger.trace(message);
 		}
 	}
+
+
+	/**
+	 * Extracts the last trusted IP address found in the X-Forwarded-For header if one is present.
+	 * Falls back to the {@link #extractDefaultRemoteAddress(ServerWebExchange)} otherwise.
+	 *
+	 * This method exists to prevent a malicious actor from spoofing the value of the X-Forwarded-For header.
+	 * If you know that your gateway application is only accessible from a a trusted load balancer,
+	 * then you can trust that the load balancer will append a valid client IP address to the X-Forwarded-For header,
+	 * and should use a value of `1` for the `maxTrustedIndex`.
+	 *
+	 *
+	 * Given the X-Forwarded-For value of [0.0.0.0, 0.0.0.1, 0.0.0.2]:
+	 *
+	 * maxTrustedIndex -> result
+	 *
+	 * [MIN_VALUE,0]   -> IllegalArgumentException
+	 * 1               -> 0.0.0.2
+	 * 2               -> 0.0.0.1
+	 * 3               -> 0.0.0.0
+	 * [4, MAX_VALUE]  -> 0.0.0.0
+	 *
+	 * @param maxTrustedIndex correlates to the number of trusted proxies expected in front spring cloud gateway
+	 *                           (index starts at 1).
+	 */
+	public static InetSocketAddress extractRemoteAddressFromXForwardedHeaderByMaxTrustedIndex(ServerWebExchange exchange, int maxTrustedIndex) {
```

Comment:
These could possible attached to an enum and therefore used in normal configuration of `RemoteAddrRoutePredicateFactory`